src/lib/y2network/udev_rule.rb

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/udev_rule_part"
+
+module Y2Network
+  # Simple udev rule class
+  #
+  # This class represents a network udev rule. The current implementation is quite simplistic,
+  # featuring a pretty simple API.
+  #
+  # @example Create a rule containing some key/value pairs (rule part)
+  #   rule = Y2Network::UdevRule.new(
+  #     Y2Network::UdevRulePart.new("ATTR{address}", "==", "?*31:78:f2"),
+  #     Y2Network::UdevRulePart.new("NAME", "=", "mlx4_ib3")
+  #   )
+  #   rule.to_s #=> "ACTION==\"add\", SUBSYSTEM==\"net\", ATTR{address}==\"?*31:78:f2\", NAME=\"eth0\""
+  #
+  # @example Create a rule from a string
+  #   rule = UdevRule.find_for("eth0")
+  #   rule.to_s #=> "ACTION==\"add\", SUBSYSTEM==\"net\", ATTR{address}==\"?*31:78:f2\", NAME=\"eth0\""
+  class UdevRule
+    class << self
+      # Returns the udev rule for a given device
+      #
+      # @param device [String] Network device name
+      # @return [UdevRule] udev rule
+      def find_for(device)
+        rules_map = Yast::SCR.Read(Yast::Path.new(".udev_persistent.net")) || {}
```

Comment:
how expensive is this call?

Reply:
It does not look expensive, but maybe it can sum up if we have dozens of cards. I will memoize the result.